bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -112,9 +112,11 @@ impl Mul<Weight> for FeeRate {
 }
 
 impl Div<Weight> for Amount {
-    type Output = FeeRate;
+    type Output = Option<FeeRate>;
 
-    fn div(self, rhs: Weight) -> Self::Output { FeeRate(self.to_sat() * 1000 / rhs.to_wu()) }
+    fn div(self, rhs: Weight) -> Self::Output {
+        self.to_sat().checked_mul(1_000).map(|s| FeeRate(s / rhs.to_wu()))
+    }
```

Comment:
This is checking for a multiplication overflow.  Although maybe it should also be performing a checked_div also in case of zero?  I'm less concerned about this right now since I'm not using it atm.  Just figured I'd add check for overflows while I'm at it.